hotsos/defs/scenarios/storage/ceph/ceph-mon/rgw_frontend.yaml

```diff
@@ -0,0 +1,29 @@
+vars:
+  rgw_frontends_from_conf: '@hotsos.core.plugins.storage.ceph.CephConfig.rgw_frontends'
+
+checks:
+  node_is_ceph_rgw_and_beast_supported_version:
+    apt:
+      librgw2:
+        - min: 14.2.0
+          max: 100.0.0
+  rgw_outdated_frontend:
+    - varops: [[$rgw_frontends_from_conf], [ne, null]]
+    - varops: [[$rgw_frontends_from_conf], [contains, civetweb]]
+  is_rgw_using_civetweb:
+    property:
+      path: hotsos.core.plugins.storage.ceph.CephCrushMap.is_rgw_using_civetweb
+conclusions:
+  rgw_outdated_frontend:
+    decision:
+      - node_is_ceph_rgw_and_beast_supported_version
+      - or:
+        - is_rgw_using_civetweb
+        - rgw_outdated_frontend
+    raises:
+      type: CephRGWWarning
+      message: >-
+        RGW is using 'civetweb' frontend. Recommend to switch to 'beast'
+        which provides better performance than civetweb; civetweb has been
+        removed from Ceph Pacific (16.2.0) onwards. Upgrading the ceph-radosgw
+        charm (version 21.10 or later) will switch to use the beast frontend.
```

Comment:
suggested ammendment:

Ceph RGW is using the 'civetweb' frontend. You are recommended to switch to 'beast'
which provides better performance (civetweb has been removed from Ceph Pacific
(16.2.0) onwards). Upgrading the ceph-radosgw charm (from stable/21.10 or newer) will
switch to use the beast frontend.